.github/workflows/build_docs.yaml

```diff
@@ -12,9 +13,13 @@ jobs:
       - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # ratchet:actions/checkout@v3
         with:
           token: ${{ secrets.DOCS_TOKEN }}
+      - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # ratchet:actions/setup-go@v3
+        with:
+          go-version: 1.18
       - name: build docs
         run: make docs
       - name: update docs
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

Comment:
@nadav-legit  I'm probably missing something here because it was already push-to-main event, and you added the pull request,
but doesn't that condition mean that the update docs job would run only in case one pushes to main? (which is never, because it's protected)